### PR TITLE
Fix weighted top_k_categorical_accuracy keras metric

### PR DESCRIPTION
**Pull request by mkysylov**
Metrics top_k_categorical_accuracy and sparse_top_k_categorical_accuracy should output a vector of accuracies instead of single scalar.

See this Keras PR for more details: https://github.com/keras-team/keras/pull/12632

### CODE REVIEW

**Comment by rthadur** [comment]
@MKisilyov please sign CLA

**Comment by MKisilyov** [comment]
I signed CLA!

**Comment by MKisilyov** [comment]
I signed it!

**Comment by pavithrasv** [comment]
Thank you for the PR. Can you add weighting tests for TopKCategorical and SparseTopKCategorical metrics here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/metrics_test.py#L864

**Comment by alextp** [comment]
I'm closing this pull request as it has way too many commits and way too many touched files. Please rebase off of head and reopen.

**Comment by MKisilyov** [comment]
@alextp Sorry for this.
Looks like GitHub didn't like that I rebased the commit without amending it.
